### ISSUE TITLE
Implement sort-by-date feature 

### ISSUE THREAD

**Issue by MarciMcGuireGCIO** [department-of-veterans-affairs/va.gov-team#17901]
## Issue Description
A new sort-by-date utility has been designed, reviewed for accessibility and design intent, and is now ready to be implemented into the Find a VA Form search results.

![](https://user-images.githubusercontent.com/75386943/104525434-c3649800-55c5-11eb-8877-69930a8105df.png)

## Tasks
- [ ] Implement accessibility recommendations provided in issue https://github.com/department-of-veterans-affairs/va.gov-team/issues/17900
- [ ] Include Google Analytics code provided in issue https://github.com/department-of-veterans-affairs/va.gov-team/issues/17273#issuecomment-751840565

## Acceptance Criteria
- [ ] Sorting utility correctly sorts results in ascending or descending order by Form Last Updated
- [ ] Google Analytics code has been implemented
- [ ] Sorting clicks are captured on staging
- [ ] Changes have been reviewed and approved by Product Owner and Product Manager on staging
- [ ] QA and Accessibility tests have been conducted on staging

@1Copenut has also reviewed this design, and believes it will meet the WCAG success criteria. He did suggest there may be an opportunity to include a more descriptive aria-label: See Slack thread:
 https://dsva.slack.com/archives/C0NGDDXME/p1608573148462600

**Comment by 1Copenut** [comment]
@MarciMcGuireGCIO One thing I forgot to mention while reviewing this the first time. What do you think about making the selected sort (A-Z in this case) black text and bold, with no underline. This helps differentiate the selected from the unselected sort.

**Comment by MarciMcGuireGCIO** [comment]
@1Copenut, that makes a lot of sense; it would be very clear how the list is currently sorted.  @johnhashva, @laurellawrence , @zacharymorel -- please see @1Copenut comment ^^.   Although he mentions A-Z, in our use case, the selected sort (would be black text, bold and no underline.  Thoughts?

**Comment by johnhashva** [comment]
@MarciMcGuireGCIO and @1Copenut -- Just to confirm, so for the following example the word _Newest_ would be bold, correct?

![image](https://user-images.githubusercontent.com/63107147/103654349-c0b5d300-4f33-11eb-8bcf-e660800945c3.png)
****

**Comment by MarciMcGuireGCIO** [comment]
@johnhashva yes, and would be also in black text as opposed to blue.

**Comment by laurellawrence** [comment]
@MarciMcGuireGCIO I'll be iterating on this and providing a few more options to choose from. I also showed this to Shawna, the VA design lead and she also had questions about scalability of this going forward, which was brought up previously. She also mentioned that it might be helpful to look at where this ticket originated... was this feedback from a previous user testing session to only provide a filter for most updated and not others? We can always implement this solution for now and provide more functionality in the future, possibly implement on the site globally. Something to think about. Stay tuned for more iterations

**Comment by MarciMcGuireGCIO** [comment]
Thanks, Laurel!  I'll try not to make a habit of doing mock-ups, BTW.  My inner UI/UX came out this morning.

There is not a great amount of detail in the GitHub issue originally created for this request in Sep https://github.com/department-of-veterans-affairs/va.gov-team/issues/17732 , and Jen is OOO this week, but I will follow up to is if she may have any additional context (email, meeting notes, etc.).

**Comment by MarciMcGuireGCIO** [comment]
Updated Tasks and AC to reflect that the sort option should only be shown when the user conducts a null search, per the context provided in the original request https://github.com/department-of-veterans-affairs/va.gov-team/issues/14112

**Comment by MarciMcGuireGCIO** [comment]
Created a new icebox issue to add sorting capability for any search once the search logic is improved upon. https://github.com/department-of-veterans-affairs/va.gov-team/issues/18184

**Comment by zacharymorel** [comment]
Is the Mock Up above the correct mock up for implementation?

**Comment by johnhashva** [comment]
@zacharymorel good catch!  Here's what we decided (https://dsva.slack.com/archives/C01DWJFD45D/p1610032516076800?thread_ts=1609945600.062500&cid=C01DWJFD45D): 

- we will implement the sorting feature as a dropdown with “Last updated” as our initial option (seen [here](https://xd.adobe.com/view/d5e7301d-12f9-438f-b30e-3646e927ed70-80b4/screen/b0086df2-db66-42e9-b202-2f790435e7c3?hints=off&x_product=cc-slack%2F1.5.0) in desktop and mobile form, per @Laurel Lawrence)

- why? (1) we can implement to meet accessibility requirements, and (2) future proof in the event we add more sorting options and (3) is more in keeping as VA.gov design pattern component.

<img width="752" alt="Screen Shot 2021-01-08 at 4 18 41 PM" src="https://user-images.githubusercontent.com/63107147/104065230-5fa92c00-51cd-11eb-80ff-d11c72736576.png">

<img width="274" alt="Screen Shot 2021-01-08 at 4 18 56 PM" src="https://user-images.githubusercontent.com/63107147/104065235-633cb300-51cd-11eb-9f9e-d2dba03340f8.png">

**Comment by zacharymorel** [comment]
Thank you @johnhashva , @MarciMcGuireGCIO , @laurellawrence . 
What is our other option of sorting besides the defaulted "Last Updated"?

**Comment by laurellawrence** [comment]
@zacharymorel, this was the reason for all the discussion. We will have a dropdown with 2 items to start for filtering with the ability to add more filters (relevance, a-z, etc.) in the future. @johnhashva and @MarciMcGuireGCIO, I believe we landed on these 2 labels with the design inspiration coming from the Gov UK site.
- Last Updated (Newest)
- Last Updated (Oldest)

**Comment by zacharymorel** [comment]
Thank you @laurellawrence !

**Comment by MarciMcGuireGCIO** [comment]
@laurellawrence when you have a minute, can you drop the XD link to the final design for the Find a VA Form search here?  Thanks!

**Comment by laurellawrence** [comment]
Final design can be found here: https://xd.adobe.com/view/e5588086-506e-49af-9202-939481ab3b5f-b00b/
<img width="953" alt="Screen Shot 2021-01-13 at 5 35 33 PM" src="https://user-images.githubusercontent.com/75386943/104525434-c3649800-55c5-11eb-8877-69930a8105df.png">

**Comment by MarciMcGuireGCIO** [comment]
Thank you!!😃

**Comment by zacharymorel** [comment]
The sorting feature should be on staging mid morning!

**Comment by MarciMcGuireGCIO** [comment]
🥳 Thanks! I'll keep an eye out!

**Comment by MarciMcGuireGCIO** [comment]
@zacharymorel, this looks great to me.  @johnhashva are you good with us getting this ready to move the Sort feature to Production?  https://staging.va.gov/find-forms/?q=10-10ez

**Comment by johnhashva** [comment]
@MarciMcGuireGCIO yes, just tested it -- and looks great.  Onward!

**Comment by zacharymorel** [comment]
Can we close this?

**Comment by MarciMcGuireGCIO** [comment]
Yes
